lib/ui/semantics.dart

```diff
@@ -0,0 +1,228 @@
+  final int index;
+
+  /// The semantics node has the quality of either being "checked" or "unchecked".
+  ///
+  /// For example, a checkbox or a radio button widget has checked state.
+  static const SemanticsFlags hasCheckedState = const SemanticsFlags._(1 << 0);
+
+  /// Whether a semantics node that [hasCheckedState] is checked.
+  ///
+  /// If true, the semantics node is "checked". If false, the semantics node is
+  /// "unchecked".
+  ///
+  /// For example, if a checkbox has a visible checkmark, [isChecked] is true.
+  static const SemanticsFlags isChecked = const SemanticsFlags._(1 << 1);
+
+  /// The possible semantics flags.
+  ///
+  /// The map's key is the [index] of the flag and the value is the flag itself.
+  static final Map<SemanticsFlags> values = const <int, SemanticsFlags>{
+    hasCheckedState.index: hasCheckedState,
+    isChecked.index: isChecked,
+  };
+}
+
+/// An object that creates [SemanticsUpdate] objects.
+///
+/// Once created, the [SemanticsUpdate] objects can be passed to
+/// [Window.updateSemantics] to update the semantics conveyed to the user.
+class SemanticsUpdateBuilder extends NativeFieldWrapperClass2 {
+  /// Creates an empty [SemanticsUpdateBuilder] object.
+  SceneBuilder() { _constructor(); }
+  void _constructor() native "SemanticsUpdateBuilder_constructor";
+
+  /// Update the information associated with the node with the given `id`.
+  ///
+  /// The semantics nodes form a tree, with the root of the tree always having
+  /// an id of zero. The `children` are the ids of the nodes that are immeidate
+  /// children of this node. The system retains the nodes that are currently
+  /// reachable from the root. A given update need not contain information for
+  /// nodes that do not change in the update. If a node is not reachable from
+  /// the root after an update, the node will be discarded from the tree.
+  ///
+  /// The `flags` are a bitfield of [SemanticsFlags] that apply to this node.
+  ///
+  /// The `actions` are a bit field of [SemanticsActions] that can be undertaken
+  /// by this node. If the user wishes to undertake one of these actions on this
+  /// node, the [Window.onSemanticsAction] will be called with `id` and one of
+  /// the possible [SemanticsActions]. Because the semantics tree is maintained
+  /// asynchronously, the [Window.onSemanticsAction] callback might be called
+  /// with an action that is no longer possible.
+  ///
+  /// The `label` is a string that describes this node.
+  ///
+  /// The `rect` is the region occupied by this node in its own coordinate
+  /// system.
+  ///
+  /// The `transform` is a matrix that maps this nodes coodinate system into its
+  /// parent's coordate system.
+  void updateNode({
+    int id,
+    int flags,
+    int actions,
+    Rect rect,
+    String label,
+    Float64List transform,
+    Int32List children
+  }) {
+    if (transform.length != 16)
+      throw new ArgumentError("[transform] must have 16 entries.");
```

Comment:
We use `[` consistently in all these ArgumentErrors.  I'll leave this for now and do a separate patch to remove them all.